docs/src/extend/custom-parsers.md

```diff
@@ -8,63 +8,54 @@ eleventyNavigation:
+ESLint custom parsers let you extend ESLint to support linting new non-standard JavaScript language features or custom syntax in your code. A parser is responsible for taking your code and transforming it into an abstract syntax tree (AST) that ESLint can then analyze and lint.
 
-* `ast` should contain the AST.
-* `services` can contain any parser-dependent services (such as type checkers for nodes). The value of the `services` property is available to rules as `context.parserServices`. Default is an empty object.
-* `scopeManager` can be a [ScopeManager](./scope-manager-interface) object. Custom parsers can use customized scope analysis for experimental/enhancement syntaxes. Default is the `ScopeManager` object which is created by [eslint-scope](https://github.com/eslint/eslint-scope).
-    * Support for `scopeManager` was added in ESLint v4.14.0. ESLint versions which support `scopeManager` will provide an `eslintScopeManager: true` property in `parserOptions`, which can be used for feature detection.
-* `visitorKeys` can be an object to customize AST traversal. The keys of the object are the type of AST nodes. Each value is an array of the property names which should be traversed. Default is [KEYS of `eslint-visitor-keys`](https://github.com/eslint/eslint-visitor-keys#evkkeys).
-    * Support for `visitorKeys` was added in ESLint v4.14.0. ESLint versions which support `visitorKeys` will provide an `eslintVisitorKeys: true` property in `parserOptions`, which can be used for feature detection.
+## Creating a Custom Parser
 
-You can find an ESLint parser project [here](https://github.com/typescript-eslint/typescript-eslint).
+If a `parseForESLint` method is exposed on the parser, this method will be used to parse the code. Otherwise, the `parse` method will be used. Both methods should take in the source code as the first argument, and an optional configuration object as the second argument (provided as `parserOptions` in a config file).
 
-```json
-{
-    "parser": "./path/to/awesome-custom-parser.js"
-}
+```javascript
+// TODO: have a simple example here w parse and parserOptions
 ```
 
-```javascript
-var espree = require("espree");
-// awesome-custom-parser.js
-exports.parseForESLint = function(code, options) {
-    return {
-        ast: espree.parse(code, options),
-        services: {
-            foo: function() {
-                console.log("foo");
-            }
-        },
-        scopeManager: null,
-        visitorKeys: null
-    };
-};
+## `parse` Return Object
 
-```
+The `parse` method should simply return the [AST](#ast-specification) object.
+
+### `parseForESLint` Return Object
```

Comment:
Maybe a bit of heading level creep here?
```suggestion
## `parseForESLint` Return Object
```

Reply:
yep! good catch. i was confused i think about the difference btwn `parse` and `parseForESLint` but the above conversation https://github.com/eslint/eslint/pull/16887#discussion_r1111065150 helped clarify.